### PR TITLE
feat(#47): add icons to distraction screen

### LINKED ISSUE #47: refactor: make the procrastination activity page look nice, with descriptions, icons, etc
DoD

- [x] It has been visually designed to be appealing to the user with clear and concise titles, descriptive icons, and other relevant details.
- [x] The page suggests a procrastination activity for the user to do at the selected location, complete with a title, description, and any other necessary information. 
- [x] The page includes a button or other element that allows the user to navigate back to the map or another page of the app once they have finished the activity.
- [x] The page has been tested thoroughly to ensure that all features and functionality are working correctly.

### PR DESCRIPTION
**Pull request by AthenaPK**
Closes #47 . Now, we can have icons associated to activities. The icon is only displayed if it exists in the database, otherwise, only name and description are displayed. Also, now there is a button at the distraction screen, allowing the user to complete the activity. For now it only displays a Toast, but later it can lead to another screen to review activity, receive points, etc. 
 I built this on top of @lipefree 's branch, since he created the DistractionScreen, so please only review the last three commits.
image

### CODE REVIEW

**Comment by aizensoosuke** [comment]
Oh my god that's so cool

File: app/src/androidTest/java/com/github/studydistractor/sdp/MainNavigationTest.kt [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
+import javax.inject.Inject
 
 @HiltAndroidTest
 class MainNavigationTest {
     @get:Rule(order = 0)
     var rule = HiltAndroidRule(this)
 
     @get:Rule(order = 1)
-    val composeRule = createComposeRule()
+    val composeRule = createAndroidComposeRule<MainActivity>()
```
Comment (aizensoosuke) [inline app/src/androidTest/java/com/github/studydistractor/sdp/MainNavigationTest.kt:27]: What is the purpose of this change?

File: app/src/androidTest/java/com/github/studydistractor/sdp/ProcrastinationActivityActivityTest.kt [hunk 2]
```diff
@@ -60,7 +60,7 @@ class ProcrastinationActivityActivityTest {
             {},
             intentFactory = {
                 val intent = Intent(it, ProcrastinationActivityActivity::class.java)
-                intent.putExtra("activity", ProcrastinationActivity(name, description, lat, long))
+                intent.putExtra("activity", Distraction(name, description, lat, long))
```
Comment (aizensoosuke) [inline app/src/androidTest/java/com/github/studydistractor/sdp/ProcrastinationActivityActivityTest.kt:63]: Probably leave that to Willy for now?

File: app/src/androidTest/java/com/github/studydistractor/sdp/history/FakeHistoryModule.kt [hunk 3]
```diff
@@ -20,12 +15,12 @@ import java.util.*
...
 class FakeHistoryModule {
     @Provides
-    fun provideHistoryInterface() : HistoryInterface{
+    fun provideFakeHistoryInterface() : HistoryInterface {
         return FakeHistoryInterface()
     }
 }
 
-class FakeHistoryInterface : HistoryInterface {
+class FakeHistoryInterface @Inject constructor(): HistoryInterface {
```
Comment (aizensoosuke) [inline app/src/androidTest/java/com/github/studydistractor/sdp/history/FakeHistoryModule.kt:23]: I strongly disagree with the name. It should be "History". [see this CodeAesthetic vid](https://www.youtube.com/watch?v=-J3wNP6u5YU)

**Review by aizensoosuke — CHANGES_REQUESTED** [review]
A really really nice idea, thanks :)

I'm curious as to why commit bb1bf30 is part of this PR though. There are a lot of changes requested on his PR, those will be reflected here if you decide to keep that commit but then you will probably have conflicts within your solutions... 
You should probably remove that commit from this branch

**Comment by AthenaPK** [comment]
> A really really nice idea, thanks :)
> 
> I'm curious as to why commit [bb1bf30](https://github.com/StudyDistractor/sdp/commit/bb1bf303ad1cab37de7bd3a062a61183c0798501) is part of this PR though. There are a lot of changes requested on his PR, those will be reflected here if you decide to keep that commit but then you will probably have conflicts within your solutions... You should probably remove that commit from this branch

So the reason is that @lipefree extended the DistractionScreen and I wrote my code on top of that. If he makes a commit with just the DistractionScreen changes, I’m happy to rebase on top of that because I agree it’s kind of messy like this. I’m open to other suggestions though.

**Comment by aizensoosuke** [comment]
Since this is a very small PR, I suggest merge last. That way you will rebase on top of willy's branch at some point.

File: app/src/androidTest/java/com/github/studydistractor/sdp/MainNavigationTest.kt (hunk 1 above)
Comment (AthenaPK) [reply app/src/androidTest/java/com/github/studydistractor/sdp/MainNavigationTest.kt:27]: This is not one of my changes, but coming from the commit I wrote my code on top of.

File: app/src/androidTest/java/com/github/studydistractor/sdp/history/FakeHistoryModule.kt (hunk 3 above)
Comment (AthenaPK) [reply app/src/androidTest/java/com/github/studydistractor/sdp/history/FakeHistoryModule.kt:23]: This is not one of my changes, but coming from the commit I wrote my code on top of.

File: app/src/androidTest/java/com/github/studydistractor/sdp/procrastinationActivity/DistractionScreenTest.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import javax.inject.Inject
+
+@HiltAndroidTest
+class DistractionScreenTest {
+    val name = "test"
```
[the remark this answers is not in the document]
Comment (AthenaPK) [reply app/src/androidTest/java/com/github/studydistractor/sdp/procrastinationActivity/DistractionScreenTest.kt:17]: This is not one of my changes, but coming from the commit I wrote my code on top of.

**Comment by AthenaPK** [comment]
Since I messed up trying to rebase, i'm opening a new PR with a fresh branch. Check out #105
